### ISSUE TITLE
RA balance

### ISSUE THREAD

**Issue by ghost** [OpenRA/OpenRA#10437]
Making a single issue for everyone to discuss RA balance for the next release due to the feedback in another thread https://github.com/OpenRA/OpenRA/pull/10432#issuecomment-169298999

The last big rebalance mainly happened due to a chance conversation between lots of people on IRC, so this may be a good place to collect everyone's views and thoughts. I'll let others start as I know some people have interesting ideas for certain issues which I am keen to hear.
##

**Comment by anjew175** [comment]
A lot people I have asked have agreed there is something off with the Allies artillery. I personally think it has to do with the range as you can have a lot of units/structures in the way of your artillery make them impossible to take care of in some situations

**Comment by cjshmyr** [comment]
Agreed wrt Artillery - Lowering its weapon range slightly from 14 (to 12?) and lowering `155mm`'s damage versus `None` very slightly could help approaches, also put the unit in more vulnerable positions. I'm assuming `None` is infantry only.

Medium & Heavy tanks - A small HP buff to these would help them "tank" better, eating an extra base defense shot or rocket. I think they should be a bit more durable and push into bases better.

MAD re-work - Differentiate it from the demo truck by removing the self destruct and letting it un-deploy. Not sure if it should become a Ukraine unit

British Spy - Get rid of this unit completely and move the normal Spy to England. I think the bonuses it grants are pretty decent. Can keep its cheap cost of $300.

Radar Dome stealth - Remove stealth detection from the radar dome. Helps phase transport and hijacker.

**Comment by ScottNZ** [comment]
:+1: Especially a tank buff

**Comment by MicroJOo** [comment]
:+1: Especially the british Spy

**Comment by Townes27** [comment]
:-1:  regarding the spies. There'd be no incentive to build dogs against Germany or France then.

**Comment by anjew175** [comment]
my suggestion for spies would be make the regular spies have a capped stealing value and dont allow them vet units, keep all other traits. This would make the British Spy a better alternative to normal spies and they are so cheaper.

**Comment by kyrylo** [comment]
Agreed wrt Artillery - :+1: but my preferred solution is https://github.com/OpenRA/OpenRA/pull/10411
Medium & Heavy tanks - :+1:
MAD re-work - dunno, this is a silly unit. Maybe make it work against structures only.
British Spy - :+1:
Radar Dome stealth - :+1: but infantry should be able to detect them (unless it already works like that)

**Comment by ShadowA** [comment]
- MAD buff #10389 Could be an interesting unit, but right now feels like not enough time/resource efficient
- Tesla Tank nerf ?  #10426
- Not sure about the whole Spy discussion :/ But I'd add #10451 (British Spy cash trick)

**Comment by pchote** [comment]
Another option for the MAD tank is to simply remove it.

**Comment by MustaphaTR** [comment]
@pchote . No. :-1:  We may increse their effect of area. I think it is 10 now. 12-15 would be better.

**Comment by ShadowA** [comment]
@pchote I also diasagree (even though nobody uses it ... yet ^_^ but I'm strating to like it ^_^ )

I think like the Demo truck it's not easy to manage / use, but it can still be an really interesting alternative to it (especially for Russia who doesn't have the Demo truck)

Maybe it should be in the description of the unit, that it has a proportional damage (or even write 33% directly), I think it's an info most people don't know about, and they get confused about this unit.

**Comment by abcdefg30** [comment]
:-1: to the spy removal from other allied factions, this removes the ability for several tactics, like powering down the enemy aa or tesla coils before an attack. What is the actual reason behind this suggestion? I didn't feel spies being over powered.

**Comment by kyrylo** [comment]
> @abcdefg30 
> this removes the ability for several tactics, like powering down the enemy aa or tesla coils before an attack

You can play as England to be able to do that, then.

> What is the actual reason behind this suggestion? I didn't feel spies being over powered.

To get rid of a duplicated unit and make factions more diverse, I think.

**Comment by Townes27** [comment]
Removing the spy from the other two factions is just stupid. Besides the obvious strategic loss there would be no way for Germany or France to detect other spies or cloaked units like the hijacker then.

**Comment by ShadowA** [comment]
There was a discussion on the forums "How to make France more appealing for players"
http://www.sleipnirstuff.com/forum/viewtopic.php?f=82&t=17666&postdays=0&postorder=asc&start=0

**Comment by MustaphaTR** [comment]
Why not add more fake structures to France like Fake Power Plants and Fake Gap Generator. I was added them to my maps, if you agree this, you can just use cameos in the map.

http://resource.openra.net/maps/6357/

**Comment by ShadowA** [comment]
@MustaphaTR Agree (especially Fake PP, great counter vs spies !)

**Comment by r34ch** [comment]
:+1: For all of @cjshmyr's points except removing spys from other allied factions. That said I do think the British spy should be replaced by some other unique ability or unit instead.

Also if the tank buffs go ahead, I think it would be a good time to add @MicroJOo's idea for balancing the turret RoT - so that med tanks have faster turret traverse than heavies.

Personally I still want to see soviets deal with air units over water. Naval maps are a chore to play 1v1 Soviets vs Allies.

**Comment by ShadowA** [comment]
@r34ch :+1: for soviet navy. 
Could be sth that : http://cnc.wikia.com/wiki/Sea_Scorpion ^_^

**Comment by cjshmyr** [comment]
> What is the actual reason behind this suggestion? 

It was to remove a duplicate unit while further incentivising choosing England, but I hear the controversy on this one.

> Removing the spy from the other two factions is just stupid. Besides the obvious strategic loss there would be no way for Germany or France to detect other spies or cloaked units like the hijacker then.

Sure there is: base defenses, radar dome, or radar jammer.

-----------

@r34ch - Agreed about navy. A couple things in mind are MiG AA, Missile Sub AA, or both. I've had the chance to play with msub's AA. I think the unit would need a cost reduction to make it accessible, and because it's a submerged unit, I had its slow-arse missile target air units. That way it was effective against helicopters or circling yaks parked over water but they could move out of the way if they were attentive. On the other hand it was a bit odd for a slow AA shot, so perhaps give it a normal rocket that explodes in a fairly short range to get the same effect.

**Comment by MustaphaTR** [comment]
Mig Really needs to be AA, Longbows are so why migs not. In fact this is another change i made in may maps. :+1: for it. Not sure about Missle Sub tho, it has really slow missiles for AG, not sure can they catch planes.

**Comment by ShadowA** [comment]
SAM sites buff ? 
They are very unbalanced against AA (lower range, rate of fire), they feel useless sometimes and frustrating

**Comment by r34ch** [comment]
@cjshmyr I would like to see how AA migs work out considering allies have the longbow.

I'm just not sure about AA msubs because I think there would have to be a compromise somewhere. If it stays late game but with AA capability it isn't solving the mid game problem and would duplicate the migs role in that case. Change the msub to be mid game and more like the destroyer and then Soviets would have to loose their long distance capability to rival the cruiser (or risk the msub being OP)


Its one of those situations where as much as it might cause controversy, for the sake of balance it could be argued that a new unit could be added. But then if that happened, I would argue allies would need a land AA equivalent to the flak - which might even help reduce the basewalking AA issue. And if that happened, you could nerf AA guns instead of buffing SAM sites @ShadowA 

That would be my dream personally, which would finally plug the 'gaps' in each sides defences and make every map playable regardless of faction.

**Comment by abcdefg30** [comment]
Another idea was to move the gunboat to the soviets as it is hardly used by allies and could get aa for the soviets (similiar to the RA2 sea scorpion). Anyway, I'm not really a fan of that, just throwing this in here. ^^

**Comment by MustaphaTR** [comment]
But soviet naval factory is a SubPen you know, (i know transport boats are not Submarine). A flak sub may be good but problem is we would need a complete new artwork for it.

**Comment by cjshmyr** [comment]
@r34ch I'd be shooting for a midgame msub, as Cruiser was always kindof akin to Mammoth in sea/land superiority, but yeah I'm really not sure yet. MiG AA could be a starting point.

**Comment by Townes27** [comment]
@cjshmyr Fair enough, forgot the radar jammer, although it costs more and requires the tech center.
Just tested some things with Gatekeeper.
Nothing other than dogs can detect spies. Other spies don't detect them either although I thought they were meant to. 

Spyplanes detect camo pillboxes. This seems odd. I can understand it revealing regular buildings, but a camouflaged pillbox is strange.

I also find it odd that you can not curtain air units even when they're on the ground.

**Comment by obrakmann** [comment]
While you're on the topic of naval balance, I'd like to draw your attention to #4502.

Personally, I'd probably make the gunboat an anti-sub/anti-ship boat only (maybe with torpedoes instead of depth charges), and remove the anti-sub capability from both the destroyer and cruiser. That would give purpose to the gunboat (would probably require a bit of a health buff to survive, though). MicroBit's water mines (sorry, can't find the issue/pr again) could be added to either the gunboat or destroyer to offset the missing stealth ability (or not, since wouldn't be canon).

**Comment by pchote** [comment]
> Personally, I'd probably make the gunboat an anti-sub/anti-ship boat only (maybe with torpedoes instead of depth charges), and remove the anti-sub capability from both the destroyer and cruiser. That would give purpose to the gunboat (would probably require a bit of a health buff to survive, though).

:+1: I've been arguing for this for literally years.

**Comment by obrakmann** [comment]
Oh, a word of caution to whoever ends up making any changes: please think of the missions when changing anything, and update them accordingly.

**Comment by cjshmyr** [comment]
fwiw, I have an old branch that follows what @obrakmann mentioned, only thing I didn't like was the torpedo missile sound on an allied ship, since I felt it's recognizable to soviet subs. Perhaps I'm a minority there ;)

**Comment by cjshmyr** [comment]
> Spyplanes detect camo pillboxes. This seems odd. I can understand it revealing regular buildings, but a camouflaged pillbox is strange.

I have no idea if that's intentional.

> I also find it odd that you can not curtain air units even when they're on the ground.

The no-IC airplanes was to prevent easy destruction past AA defenses

**Comment by pchote** [comment]
> only thing I didn't like was the torpedo missile sound on an allied ship since I felt it's iconic to soviet subs.

Can we get a different sound (and maybe a different bubble animation?) for it?

**Comment by Micr0Bit** [comment]
@obrakmann : 
![mines](https://cloud.githubusercontent.com/assets/9139965/12185373/e23fe5d4-b59a-11e5-8395-91c336a7f769.png)

**Comment by GraionDilach** [comment]
If sounds are the problem, then a TD/TS/pitchshifted RA sound might do the trick regarding the torpedo sound.

**Comment by r34ch** [comment]
I take back my thoughts on the spy earlier; I think giving the spy exclusively to England is a good idea. The phase transport is geared towards this and it won't break any existing strategies that you can't already do if you simply choose England.

:+1: Really good suggestions @cjshmyr 

As for the torpedo sound for the PT boat, I think the existing sound is fine imho. When I think soviet subs I think of the double launch. I would imagine PT boats would just fire a single torp so that alone would make the overall sound 'different' enough.

**Comment by ShadowA** [comment]
> Why not add more fake structures to France like Fake Power Plants and Fake Gap Generator. I was added them to my maps, if you agree this, you can just use cameos in the map.

@MustaphaTR add to your ideas : 
- Fake Refinery, to deter usage of spies for stealing
- You proposed Fake Gap. Maybe Fake defences, to frighten the enemy ?
- And fake Tanks ^_^ (to deceive the enemy with a fake army / who could for instance be slower) ... but I think that's too much ^_^

**Comment by abcdefg30** [comment]
Fake Tanks sound like a funny idea, in fact they were used in reality too ([Dummy tank - wikipedia](https://en.wikipedia.org/wiki/Dummy_tank)).

But I'm not sure if they are fitting into red alert, though.

**Comment by ShadowA** [comment]
@abcdefg30 Yes ! :)

Actually I know only 1 RTS game that uses them (http://ruse.wikia.com/wiki/Ruses)
(never played it though)

> Decoy Offensive: This ruse allows the player to create decoy units to simulate an army. Decoy units are produced and then launched in this sector in order to simulate an attack. Any units created are only those which have been researched. Decoys are as fast as their real counterparts, but cannot attack and will be destroyed with one shot from any unit that can damage it.

But maybe that's too much for France, and would be new to RA (but why not ? :) )

**Comment by SoScared** [comment]
:-1: Artillery range nerf. I don't recognize the argument for this. The Artillery gets to be OP under certain conditions: Exclusively choke maps and/or matches between passive players. I don't think that's enough reason to reduce the distinctiveness between the factions.

I think a view range nerf sounds reasonable as in #10411

:8ball: Heavy/Medium tanks health buff. I'd be extra careful with this. I'm under the impression that tanks are sufficiently strong for their purpose (tanking damage). I've been for slight tank buffs in the past just to see them more on the battlefield. It looks currently that they have made a return to the meta without any buffs.

:+1: Missile Sub AA. I don't want to see the Allied air dominance equalized in the late game. The Missile Sub AA will simply deter Allies from using stationary heli's as passive spotters and play whack-a-mole with the Missile Sub without risk. I was originally for another weapon as AA to make it more effective but after watching a demonstration, using the current weapon seemed preferable (slow, double missile) and would serve (my idea of) it's purpose.

:-1: MiG AA. This would give Longbows a greater role. You really don't see the Longbow any more. This would greatly deter Allies from building (slow) Hinds as they would become easy prey for the Migs, and bear in mind, Allies don't have an effective mobile AA on the field. I'd imagine you'd need a slightly buffed Longbow (discussed earlier: faster rockets/increased HP/price reduction) so that the Longbow becomes more than just a Hind-babysitter. I also imagine the Hind would possibly have to be altered (e.g. increased speed). I'd give it an :8ball: with at least an alteration of the Longbow.

I want to say I protest arguments that rests on "for the sake of balance" without weighing in the consequences. There's no reason for the factions to be more equal if it doesn't improve gameplay.

+1 vision range for the Light Tank. Gives it a better scout and infantry support role (same as Soviet dogs). It currently has the same view range as basic infantry. When/if husks eventually can be destroyed in the ore ref in future releases, the current Light Tank will be utterly useless. +1 view range will change that. Also discussed, higher turret rotation for better use against other light vehicles.

-1 vision range (minimum) for the Transport helicopter. The view range was reduced from 12 to 10 from the previous release. I understand the view range buff initially was used to encourage more use of it. Now I want to see it's view range lowered below the Hind's. I want to see it further discouraged for scouting and currently there's close to 0 risk of loosing it when dropping in and around your opponent's base because of it's view range.

**Comment by penev92** [comment]
Calm down, guys. The current release just came out. And it contained a lot of balance changes over the previous one. I bet the meta hasn't even settled yet. Please wait for that to happen before starting with random balance changes all across the mod. Thanks.

**Comment by cjshmyr** [comment]
@SoScared Nice input!

Those light tank changes were merged in recently. As for MiG AA I agree on your points. Also would want to take into consideration `MRJ` for countering it, but no idea how effective it would be. 

Artillery: I've gone from loving to absolutely despising this unit. I agree it should retain its identity of having a longer range than any other land unit (especially compared to its soviet counterpart), but to me the unit encourages base creep or passive play.

**Comment by anjew175** [comment]
It is worth looking at the TD artillery. It has almost half the range and more damage vs buildings. A balance between the current RA artillery and TD artillery would make it seem like less of an OP unit.

Also a lot of the changes being suggested arent related to any of the recent changes. The recent changes just affect most faction units, units which you rarely see in a 1v1

**Comment by Townes27** [comment]
@cjshmyr 
I don't think artillery encourages base creep but the lack of mobile AA does, especially since suicide yaks are back.

**Comment by klaas1** [comment]
+1 for everything SoScared said.

I like the current balance, both factions have their strongs and weaknesses. Only two units that are currently under-represented are MiGs and longbows. If MiGs are not given AA capabilities, I think they should be stronger vs. ground units. They take loads of recourses, are micro intensive, but are not giving a real punch IMHO.

Maybe increasing the damage, but reducing the number of rockets would be an option? This way it would be the ideal hit and run unit. When MiGs are flying around more often, maybe longbows will be more useful too. It would be nice if Longbows get a small buff vs armored units though, at the moment Hinds do more damage to about every target, which reduces Longbows to expensive AA.

**Comment by MurtoTheRay** [comment]
+1 for moving spies to Britain and what SoScared said.

Also: What about mammoth tanks? They arent used much and arent really an upgrade from the heavy tanks as they are far too slow and fire too slowly, how about some love in their department?

**Comment by StexoO** [comment]
@MurtoTheRay wt... Not really an upgrade from the heavy tanks ? oO

:+1: Mammoth tanks are __extremely__ slow (to build and to move) and __extremely__ expensive __BUT__
- Best tanking
- Anti-infantry
- Anti-air
- Can get rid of one defense alone (what the defense is doesn't matter)
- Self-repair until 50 % (Self-repair until 50% !)
- (Can be curtained)
- Almost nuke proof if it's not in the center of (or one cell to the center of) the impact ! (And repairs itself after being hit by a nuke, of course...)

I vote for a self-repair to 100 %, so they really worth it. :yum:

**Comment by kyrylo** [comment]
My main issue with mammoth tanks in RA is that once you attack there's no way back. They're too slow to retreat.

**Comment by MurtoTheRay** [comment]
@StexoO You can get a lot more done with heavy tanks + infantry than you can with mammoths + infantry. Heavy tanks are faster therefore they can respond quicker to enemy attacks and they can run over infantry easily due to their speed which amounts to way more damage than their main weapon.

Mammoths tend to die to infantry before they get any decent crushes and will then block your units with their husks. Anti-air is only useful if your opponent has plenty of air for you to take down and air isnt good vs mammoths so there is no reason for air to be near them.

On the other hand, self-healing is great; would love to see it up to 100%. I just don't see a place for mammoths except on linear maps such as singles or vegetation where they can be used to steam roll through enemy defenses.

**Comment by MustaphaTR** [comment]
What about adding onmicrushing to Mammoth Tank. This is another change i did in my maps.

From map description:
> ...Mammoths can crush Kennel, PillBoxes, Brick Wall, Flame Turret, Silo, Ranger, Light Tank, Mobile Gap Gen., V2, Artillery, Mobile Flak, Supply Truck and Demo. Truck.

**Comment by MurtoTheRay** [comment]
I thiink making it so that mammoths could crush concrete walls  would be nice. Maybe they shouldnt be slowed down by terrain either?

**Comment by klaas1** [comment]
Omni-crushing mammoths +1. Finally a way to literally steamroll fortified players. Although it probably needs some polish + explosions to make it look decent.

**Comment by hacikenks** [comment]
fake powerplant :+1:
shorter arty vision :+1:
perhaps also longer build time + higher cost for arties

**Comment by HenrytheSlav** [comment]
fake powerplants :+1:
changing arty :-1: right now, maybe slightly lower range
Omni-crushing mammoths :+1: seems to bring them back in game
migs AA definitely :-1: as this gives soviets easy pickings against helis who are clumsy compared to them, plus you can't even sneakily attack enemy as he will have almost instant response unit

**Comment by anjew175** [comment]
Just going to copy what i said on the forum:
"I implore people to look at the arty from TD, it has a significantly smaller range but more damage and it works for its situational work. Im not suggesting to use TD stats but it shows that you dont need the incredibly massive range the arty currently has."

The current arty range is incredible when you compare mods and sure they are "completely" different mods it still doesnt make sense why RA needs almost double TD's range. It promotes leaving your units in your base and just focusing on vision. You can also nerf V2 range if you want to keep that particular aspect of their distinctiveness however I think the fact one shoots missles and one shoots artillery bullets is the distinctiveness as they both do more damage to particular targets.

**Comment by MurtoTheRay** [comment]
After thinking for a while i think that Anjew is spot on here. There is no reason why arty should have a much greater range than a V2. V2s are better at taking out buildings due to their supreme accuracy, but against units the artillery rains supreme; this is balanced enough but the fact that arty has a larger range skews it to make arty quite a bit better. This needs some more testing though; its a big change.

**Comment by MustaphaTR** [comment]
Fake PPs for France looks got a lot of :+1:s here. I'll open a PR for it when i back home in 2 hours. We may discuss more abour it at there.

**Comment by SoScared** [comment]
Adding Mammoth Speed buff to the discussion #11077

Also discussed at: http://www.sleipnirstuff.com/forum/viewtopic.php?f=82&p=295320#295320

Mammoth Speed buff from 42 to 50 would give it almost a 20% speed increase. Enough to encourage playtesting, not enough to make it a monster unit. Would love to have this played out.

**Comment by r34ch** [comment]
LB and Mig AG weapon buffs #11337

**Comment by abcdefg30** [comment]
This is already quite old now, and balancing changed a lot (see #12739 for the next playtest). Discussion seems to happen mostly on the forums today anyway.

**Comment by Mailaender** [comment]
This has a small bounty https://www.bountysource.com/issues/29648355-ra-balance I suggest @SoScared claims it for his long term commitment.
